### ISSUE TITLE
sql: TestCCLLogic/fakedist-metadata/partitioning_enum is flaky

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#75227]
This test flakes in CI and nightlies since https://github.com/cockroachdb/cockroach/issues/68395#issue-959452230.

```
$ make stress PKG='./pkg/ccl/logictestccl' TESTS='TestCCLLogic/fakedist-metadata/partitioning_enum' TESTFLAGS='' TESTTIMEOUT=30s

0 runs so far, 0 failures, over 5s
22 runs so far, 0 failures, over 10s

I220120 17:17:30.587039 1 (gostd) rand.go:147  [-] 1  random seed: -3836311176851664413
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /var/folders/3j/7c3w0jjd14g4n19lvvrxtm2r0000gp/T/logTest
CCLLogic2338896894
    test_log_scope.go:80: use -show-logs to present logs inline
=== RUN   TestCCLLogic/fakedist-metadata
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-metadata/partitioning_enum
=== CONT  TestCCLLogic/fakedist-metadata/partitioning_enum
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value
    logic.go:1985:

        /Users/marcus/go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/pa
rtitioning_enum:92:
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@
idx", but no error occurred
--- done: /Users/marcus/go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/
partitioning_enum with config fakedist-metadata: 15 tests, 1 failures
--- total progress: 15 statements/queries
--- total: 15 tests, 1 failures
=== CONT  TestCCLLogic
    logic.go:3604: -- test log scope end --
test logs left over in: /var/folders/3j/7c3w0jjd14g4n19lvvrxtm2r0000gp/T/logTestCCLLogic2338896894
--- FAIL: TestCCLLogic (2.94s)
    --- FAIL: TestCCLLogic/fakedist-metadata (0.00s)
        --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum (2.84s)
            --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value (1.07
s)
FAIL

ERROR: exit status 1

40 runs completed, 1 failures, over 14s
context canceled
FAIL
FAIL    github.com/cockroachdb/cockroach/pkg/ccl/logictestccl   13.886s
FAIL
gmake: *** [Makefile:1087: stress] Error 1
```

**Comment by irfansharif** [comment]
Should this be closed after #75300? +cc @ajwerner.

**Comment by ajwerner** [comment]
Yes
